master/Master.hs

```diff
@@ -132,45 +96,40 @@ sendResponse (KVVote tid sid VoteReady request) = do
   else if timeout then sendDecisionToRing (KVDecision tid DecisionAbort request)
        else return ()
 
-sendResponse kvMsg@(KVResponse tid sid _) = ask >>= \mvar -> liftIO $ do
-  state <- takeMVar mvar
-  -- use ackMap to keep track of which transactions on the GET pathway have been
-  -- forwarded to the client
-  let oldAckMap = ackMap state
-
-  case Map.lookup tid oldAckMap of
-    Nothing -> do
-      let newAckMap = Map.insert tid (Set.singleton sid) oldAckMap
-
-      putMVar mvar $ state {ackMap = newAckMap} --MasterState (socket state) (cfg state) (voteMap state) newAckMap
-
-      sendMsgToClient kvMsg (cfg state)
-
-    Just hasRespondedSet  -> do
-      let hasRespondedSet' = Set.insert sid hasRespondedSet
-          newAckMap' = if Set.size hasRespondedSet' == (Prelude.length $ slaveConfig $ cfg state)
-                       --erase this txn from the ack map, all the slaves have
-                       --sent a response to master
-                       then Map.delete tid oldAckMap
-                       else Map.insert tid hasRespondedSet' oldAckMap
-
-      putMVar mvar $ state { ackMap = newAckMap'} --MasterState (socket state) (cfg state) (voteMap state) newAckMap'
-
-sendResponse kvMsg@(KVRequest txn_id req) = ask >>= \mvar -> do
+processMessage kvMsg@(KVResponse tid sid _) = do
+  addAck tid sid
+  complete <- ackComplete tid
```

Comment:
i see, so right now we also should be waiting to forward to the client until all `GETS` are received? (i.e. we are handling `GET`s and `PUT`s in the exact same way w.r.t. timeout). So there shouldn't be any weird errors on `GET`s anymore (i.e. master timing out `GET`s that have already been forwarded to client). Is that correct?